yapapi/executor/__init__.py

```diff
@@ -508,19 +508,19 @@ async def worker_starter() -> None:
                     "golem.requestor.code": "Success",
                 }
 
+            if workers:
+                logger.log(log_level, "Waiting for %d workers to finish...", len(workers))
+                await asyncio.wait(
+                    workers.union(services), timeout=10, return_when=asyncio.ALL_COMPLETED
+                )
+
```

Comment:
It's better to first let the workers finish their work and then terminate the corresponding agreements. Doing this the other way round will cause the `asyncio` tasks for the workers that still run to be cancelled which will produce warning messages in the debug log.